docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2023-3",
+        "name": "BlockDeduplicatorVerbatimAssemblyItemBrokenComparison",
+        "summary": "Two blocks that are completly identical except for a difference in the data of a ``verbatim`` assembly item may wrongfully compare equal.",
+        "description": "The block deduplicator is an optimizer step which identifies equivalent assembly blocks and merges them into a single one. However when blocks contained ``verbatim`` data, their comparison was wrongfully executed, leading to the collapsing of assembly blocks which are identical but on the data of ``verbatim`` items. This could produce final bytecode which does not run as the intended solidity code. The bug can be avoided, in versions previous to the fix, by disabling the flag ``deduplicate`` in the optimizer details settings via the standard Json input interface.",
```

Comment:
```suggestion
        "name": "VerbatimInvalidDeduplication",
        "summary": "``verbatim`` calls with different data can wrongfully be de-duplicated if used in very similar context.",
        "description": "The block deduplicator is an optimizer step which identifies equivalent assembly blocks and merges them into a single one. However when blocks contained ``verbatim`` data, their comparison was implemented incorrectly, leading to the collapse of assembly blocks which are identical except on the data of ``verbatim`` items. Since verbatim is only available in Yul, compilation of Solidity sources is not affected.",
```